File: CSharp/Tests/RepeatingLettersNthTimesTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class RepeatingLettersNthTimesTest
    {
        [Theory]
        [InlineData("mice", 5, "mmmmmiiiiiccccceeeee")]
        [InlineData("hello", 3, "hhheeellllllooo")]
        [InlineData("stop", 1, "stop")]
        [InlineData("tomato", 2, "ttoommaattoo")]
        public void Repeat_StringInputValue_ReturnStringWithCharactersRepeatedNthTimes(string str, int num, string expected)
        {
            var actual = RepeatingLettersNthTimes.Repeat(str, num);

            Assert.Equal(expected, actual);
        }
    }
}

File: js/tests/repeatingLettersNthTimes.test.js
const repeat = require("../repeatingLettersNthTimes");

const data = [
  ["mice", 5, "mmmmmiiiiiccccceeeee"],
  ["hello", 3, "hhheeellllllooo"],
  ["stop", 1, "stop"],
  ["tomato", 2, "ttoommaattoo"]
];

describe("repeat", () => {
  describe("Return string with repeated charcters nth times", () => {
    test.each(data)("repeat(%p, %i) = %p", (str, num, result) => {
      var actual = repeat(str, num);

      expect(actual).toEqual(result);
    });
  });
});
